src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts

```diff
@@ -3167,208 +3167,647 @@ describe('SQL Builder test', () => {
-        device.operating_system::varchar as device_operating_system,
-        device.operating_system_version::varchar as device_operating_system_version,
-        device.ua_browser::varchar as device_ua_browser,
-        device.ua_browser_version::varchar as device_ua_browser_version,
-        device.ua_os::varchar as device_ua_os,
-        device.ua_os_version::varchar as device_ua_os_version,
-        device.ua_device::varchar as device_ua_device,
-        device.ua_device_category::varchar as device_ua_device_category,
-        device.system_language::varchar as device_system_language,
-        device.time_zone_offset_seconds::bigint as device_time_zone_offset_seconds,
-        device.advertising_id::varchar as device_advertising_id,
-        geo.continent::varchar as geo_continent,
-        geo.country::varchar as geo_country,
-        geo.city::varchar as geo_city,
-        geo.metro::varchar as geo_metro,
-        geo.region::varchar as geo_region,
-        geo.sub_continent::varchar as geo_sub_continent,
-        geo.locale::varchar as geo_locale,
-        platform,
-        project_id,
-        traffic_source.name::varchar as traffic_source_name,
-        traffic_source.medium::varchar as traffic_source_medium,
-        traffic_source.source::varchar as traffic_source_source,
-        user_first_touch_timestamp,
-        user_id,
-        user_pseudo_id,
-        user_ltv,
-        event_dimensions,
-        ecommerce,
-        items,
-        TO_CHAR(
-          TIMESTAMP 'epoch' + cast(event_timestamp / 1000 as bigint) * INTERVAL '1 second',
-          'YYYY-MM'
-        ) as month,
-        TO_CHAR(
-          date_trunc(
-            'week',
-            TIMESTAMP 'epoch' + cast(event_timestamp / 1000 as bigint) * INTERVAL '1 second'
-          ),
-          'YYYY-MM-DD'
-        ) || ' - ' || TO_CHAR(
-          date_trunc(
-            'week',
-            (
+    with
+      tmp_data as (
+        select
+          event_date,
+          event_name,
+          event_id,
+          event_bundle_sequence_id::bigint as event_bundle_sequence_id,
+          event_previous_timestamp::bigint as event_previous_timestamp,
+          event_server_timestamp_offset::bigint as event_server_timestamp_offset,
+          event_timestamp::bigint as event_timestamp,
+          ingest_timestamp,
+          event_value_in_usd,
+          app_info.app_id::varchar as app_info_app_id,
+          app_info.id::varchar as app_info_package_id,
+          app_info.install_source::varchar as app_info_install_source,
+          app_info.version::varchar as app_info_version,
+          device.vendor_id::varchar as device_id,
+          device.mobile_brand_name::varchar as device_mobile_brand_name,
+          device.mobile_model_name::varchar as device_mobile_model_name,
+          device.manufacturer::varchar as device_manufacturer,
+          device.screen_width::bigint as device_screen_width,
+          device.screen_height::bigint as device_screen_height,
```

Comment:
Add tests for the common attribute `device.viewport_width` and `device.viewport_height` for the Web SDK